cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CopyNativeLibTask.java

```diff
@@ -74,37 +78,45 @@ public CopyNativeLibTask() {
     
     @Override
     public void execute(BuildContext buildContext) {
+        CompilerContext context = buildContext.get(BuildContextField.COMPILER_CONTEXT);
+        packageCache = PackageCache.getInstance(context);
+        skipTests = buildContext.skipTests();
 
         Path sourceRootPath = buildContext.get(BuildContextField.SOURCE_ROOT);
         String balHomePath = buildContext.get(BuildContextField.HOME_REPO).toString();
         this.manifest = ManifestProcessor.getInstance(buildContext.get(BuildContextField.COMPILER_CONTEXT)).
                 getManifest();
-        List<BLangPackage> moduleBirMap  = buildContext.getModules();
-        if (buildContext.getSourceType() == SINGLE_BAL_FILE) {
-            copyImportedJarsForSingleBalFile(buildContext, moduleBirMap, sourceRootPath, balHomePath);
-            return;
-        }
-        copyImportedJarsForModules(buildContext, moduleBirMap, sourceRootPath, balHomePath);
-    }
-
-    private void copyImportedJarsForSingleBalFile(BuildContext buildContext, List<BLangPackage> moduleBirMap,
-                                                  Path sourceRootPath, String balHomePath) {
-        // Iterate through the imports and copy dependencies.
-        HashSet<PackageID> alreadyImportedSet = new HashSet<>();
-        for (BLangPackage pkg : moduleBirMap) {
-            copyImportedLibs(pkg.symbol.imports, buildContext.moduleDependencyPathMap.get(pkg.packageID).platformLibs,
-                             buildContext, sourceRootPath, balHomePath, alreadyImportedSet);
-        }
+        copyImportedJarsForModules(buildContext, buildContext.getModules(), sourceRootPath, balHomePath);
     }
 
-    private void copyImportedJarsForModules(BuildContext buildContext,  List<BLangPackage> moduleBirMap,
+    private void copyImportedJarsForModules(BuildContext buildContext, List<BLangPackage> moduleBirMap,
                                             Path sourceRootPath, String balHomePath) {
         // Iterate through the imports and copy dependencies.
         HashSet<PackageID> alreadyImportedSet = new HashSet<>();
         for (BLangPackage pkg : moduleBirMap) {
-            // Copy jars from imported modules.
-            copyImportedLibs(pkg.symbol.imports, buildContext.moduleDependencyPathMap.get(pkg.packageID).platformLibs,
-                             buildContext, sourceRootPath, balHomePath, alreadyImportedSet);
+            PackageID packageID = pkg.packageID;
+            BLangPackage bLangPackage = packageCache.get(packageID);
+
+            if (bLangPackage == null || !buildContext.moduleDependencyPathMap.containsKey(packageID)) {
```

Comment:
When will bLangPackage be null ?

Reply:
Can be, if it is not there in the package cache.